doc/semantics-of-the-publishing-api.md

```diff
@@ -0,0 +1,406 @@
+against the user’s request URL when determining whether a redirect should occur.
+A *prefix type* denotes that any subpath under the specified path should be
+redirected to the destination. You can think of these as “exact” and “fuzzy”
+matches, respectively.
+
+The `redirects` are generally optional, except for the case when the content
+item has a `format` of *redirect*. In this case, the redirects must be present
+and they must include the `base_path` of content_item in the *path* property, at
+the very least.
+
+Redirects are subject to the same requirement as routes in that their paths must
+reside under the `base_path` of the content item (see [**routes**](#routes)).
+
+Note: Collectively, routes and redirects must have unique paths. The Publishing
+API will not accept content items where the routes and redirects conflict with
+each other.
+
+---
+
+### update_type
+
+Example: *major*
+
+Required: Conditionally
+
+The `update_type` is an indicator of how much the content item has changed since
+the last time it was published. This field is required when publishing the
+content item.
+
+There’s no restriction on what the `update_type` should be, but at present we
+use:
+
+- *minor* - for minor edits that probably won’t be of interest to users
+
+- *major* - for major edits that probably will be of interest to users
+
+- *republish* - for when the content item needs to be re-sent to downstream
+systems
+
+- *links* - for when only the links of the content item have changed
+
+An example case for when a republish `update_type` should be used is when there
+had previously been a problem with a downstream system and that system did not
+act on publish events correctly. For example, there could have been a problem
+with sending email alerts and so one way that this could be rectified might be
+to republish the content items.
+
+The *links* `update_type` is set automatically when the /links endpoint is used.
+There is no need to set this manually. All of these `update_types` form part of
+the routing key when the content item document is placed on the message queue,
+together with the `format` of the content item (e.g. *policy.major*)
+
+---
+
+### **Optional Fields**
+
+---
+
+### description
+
+Example: *VAT rates for goods and services*
+
+Required: No
+
+The `description` captures a short explanation of the content item. It is sent
+downstream to the content store and may be displayed at the discretion of the
+front-end application, when the content item is rendered.
+
+---
+
+### need_ids
+
+Example: *["1234", "1235"]*
+
+Required: No
+
+The `needs_ids` are the identifiers of [user needs](https://www.gov.uk/design-principles)
+that are entered through the [Maslow Application](https://github.com/alphagov/maslow).
+They are passed through to the content store, untouched by the pipeline. The
+front-end applications can then use the `needs_ids` to present pages to users,
```

Comment:
This sentence doesn't provide any actual information about what type of pages are presented without clicking the link.